pkg/agent/flowexporter/connections/connections.go

```diff
@@ -234,14 +253,22 @@ func (cs *connectionStore) ForAllConnectionsDo(callback flowexporter.ConnectionM
 func (cs *connectionStore) Poll() ([]int, error) {
 	klog.V(2).Infof("Polling conntrack")
 	// Reset isActive flag for all connections in connection map before dumping flows in conntrack module.
-	// This is to specify that the connection and the flow record can be deleted after the next export.
-	resetConn := func(key flowexporter.ConnectionKey, conn flowexporter.Connection) error {
-		conn.IsPresent = false
-		cs.connections[key] = conn
+	// if the connection does not exist in conntrack table and exported, we will delete it from connection map.
+	deleteIfStaleOrResetConn := func(key flowexporter.ConnectionKey, conn flowexporter.Connection) error {
+		if !conn.IsPresent && conn.DoneExport {
+			if err := cs.deleteConnectionByKeyWithoutLock(key); err != nil {
+				return err
+			}
+		} else {
+			conn.IsPresent = false
+			cs.connections[key] = conn
```

Comment:
this copy seems needlessly "expensive" to me
do you know why `ForAllConnectionsDo` works with a value and not a pointer? Or maybe `ForAllConnectionsDo` is not the right solution for this operation

Reply:
Switching to using a pointer for `ForAllConnectionsDo` is fine to me in this case. Tagging @srikartati  here to see whether he has other concerns.

Reply:
Switching to pointer for value in `ForAllConnectionsDo` of connections.go works for me. I do not see any issues with other usages of this method. Delete/reset connections can be definitely made efficient.